pkg/storage/gc/gc.go

```diff
@@ -366,6 +388,11 @@ func (gc GarbageCollect) removeTagsPerRetentionPolicy(ctx context.Context, repo
 		return nil
 	}
 
+	// skip the global blobs repo
+	if repo == constants.GlobalBlobsRepo {
```

Comment:
Something is fishy.
Why is this here? `constants.GlobalBlobsRepo` is not an actual repo because it has an invalid repo name. How is GC detecting it when "walking" the disk if it doesn't have a valid repo name? It shouldn't call this function at all for `constants.GlobalBlobsRepo`?